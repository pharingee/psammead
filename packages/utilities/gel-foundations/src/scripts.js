export arabic from './scripts/arabic';

export arabicPashto from './scripts/arabic-pashto';

export bengali from './scripts/bengali';

export burmese from './scripts/burmese';

export devanagariAndGurmukhi from './scripts/devanagari-and-gurmukhi';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';

export cyrillic from './scripts/latin-and-cyrillic';
export cyrillicAndLatin from './scripts/latin-and-cyrillic';
export latin from './scripts/latin-and-cyrillic';

export latinDiacritics from './scripts/latin-with-diacritics';

export chinese from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export noAscendersOrDescenders from './scripts/no-ascenders-or-descenders';

export sinhalese from './scripts/sinhalese';

export tamil from './scripts/tamil';

export thai from './scripts/thai';
