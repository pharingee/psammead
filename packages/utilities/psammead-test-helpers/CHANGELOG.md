# Psammead Test Helpers Changelog

<!-- prettier-ignore -->
| Version | Description |
|---------|-------------|
| 3.0.2 | [PR#2166](https://github.com/bbc/psammead/pull/2166) remove async keyword from test helper to fix regenerator runtime plugin error |
| 3.0.1 | [PR#2123](https://github.com/bbc/psammead/pull/2125) Removes use of async/await |
| 3.0.0 | [PR#2104](https://github.com/bbc/psammead/pull/2104) add function to detect and render helmet for use in snapshots |
| 2.0.2 | [PR#1994](https://github.com/bbc/psammead/pull/1994) Add comment to clarify new shouldMatchSnapshot logic |
| 2.0.1 | [PR#1958](https://github.com/bbc/psammead/pull/1958) Update `shouldMatchSnapshot` to add firstChild logic |
| 2.0.0 | [PR#1917](https://github.com/bbc/psammead/pull/1917) Remove react-test-renderer and shallow rendering |
| 1.3.2 | [PR#1919](https://github.com/bbc/psammead/pull/1919) Accidental version bump |
| 1.3.1 | [PR#1849](https://github.com/bbc/psammead/pull/1849) Update README.md |
| 1.3.0 | [PR#1846](https://github.com/bbc/psammead/pull/1846/) Import test helper functions from simorgh |
| 1.2.1 | [PR#1803](https://github.com/bbc/psammead/pull/1803/) Patches broken links on badges in documentation |
| 1.2.0 | [PR#1794](https://github.com/bbc/psammead/pull/1794) Add david dependency badges |
| 1.1.1 | [PR#1682](https://github.com/bbc/psammead/pull/1682) Move all dev dependencies to top level package.json |
| 1.1.0   | [PR#1233](https://github.com/bbc/psammead/pull/1233) Add ESM modules entry |
| 1.0.2   | [PR#1083](https://github.com/bbc/psammead/pull/1083) Use `jest-styled-components@6.3.3` |
| 1.0.1   | [PR#732](https://github.com/bbc/psammead/pull/732) Use @testing-library/react for generating snapshots |
| 1.0.0   | [PR#679](https://github.com/bbc/psammead/pull/679) Bump version number |
| 0.3.4   | [PR#424](https://github.com/bbc/psammead/pull/424) Add Snyk badge to readme |
| 0.3.3   | [PR#323](https://github.com/BBC/psammead/pull/323) Update readme storybook badge |
| 0.3.2   | [PR#264](https://github.com/BBC/psammead/pull/319) Resolving package-lock issues. |
| 0.3.1   | [PR#264](https://github.com/BBC/psammead/pull/264) Resolving package-lock issues. |
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
