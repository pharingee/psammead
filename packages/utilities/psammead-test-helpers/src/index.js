import { render } from '@testing-library/react';
import 'jest-styled-components';
import deepClone from 'ramda/src/clone';
import renderWithHelmet from './renderWithHelmet';

export const shouldMatchSnapshot = (title, component) => {
  it(title, done => {
    // select the first child to remove the pointless wrapping div from snapshots
    const removeWrappingDiv = container => container.firstChild;
    renderWithHelmet(component).then(({ container }) => {
      const hasOneChild = container.children.length === 1;
      /*
       * if the container has more than one child then it's a component that uses a
       * fragment at the top level so we should not select the first child because it
       * wouldn't snapshot the whole component
       */
      expect(
        hasOneChild ? removeWrappingDiv(container) : container,
      ).toMatchSnapshot();

      done();
    });
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const { container } = render(component);
    expect(container.firstChild).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

export const suppressPropWarnings = warnings => {
  const { expectedWarnings } = window;
  if (expectedWarnings && Array.isArray(expectedWarnings)) {
    window.expectedWarnings = [...expectedWarnings, warnings];
  } else {
    window.expectedWarnings = [warnings];
  }
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
