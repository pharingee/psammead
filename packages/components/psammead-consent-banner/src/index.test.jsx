import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { arabic, latin } from '@bbc/gel-foundations/scripts';
import { ConsentBanner, ConsentBannerText } from '.';

const baseProps = {
  accept: <button type="button">Accept</button>,
  dir: 'ltr',
  id: 'banner-id',
  reject: <a href="https://foobar.com">Reject</a>,
  script: latin,
  service: 'news',
  text: <p>Hello</p>,
  title: "We've updated our Privacy and Cookies Policy",
};

const rtlProps = {
  accept: <button type="button">قبول</button>,
  dir: 'rtl',
  id: 'banner-id',
  reject: <a href="https://foobar.com">رفض</a>,
  script: arabic,
  service: 'arabic',
  text: <p>مرحبا</p>,
  title: 'عنوان',
};

describe('ConsentBanner', () => {
  shouldMatchSnapshot(
    'should correctly render for ltr service',
    <ConsentBanner {...baseProps} />,
  );

  shouldMatchSnapshot(
    'should correctly render for rtl service',
    <ConsentBanner {...rtlProps} />,
  );

  describe('with hidden attribute on wrapper', () => {
    const props = {
      hidden: true,
      ...baseProps,
    };

    shouldMatchSnapshot(
      'should correctly render',
      <ConsentBanner {...props} />,
    );
  });
});

describe('ConsentBannerText', () => {
  shouldMatchSnapshot(
    'should correctly render',
    <ConsentBannerText dir="ltr" script={latin} service="news">
      We have made some important changes to our Privacy and Cookies Policy and
      we want you to know what this means for you and your data.
    </ConsentBannerText>,
  );
});
