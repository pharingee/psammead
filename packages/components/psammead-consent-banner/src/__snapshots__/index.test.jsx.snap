// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentBanner should correctly render for ltr service 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  background-color: #323232;
  padding: 1rem 0.5rem;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c1::after {
  content: '\\0020';
  display: block;
  height: 0;
  clear: both;
  overflow: hidden;
  visibility: hidden;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #FFFFFF;
  font-weight: 700;
  padding: 0;
  margin: 0;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #F6A21D;
  font-weight: 600;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c3 li + li {
  padding-top: 0.5rem;
}

.c4 button {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #F6A21D;
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
}

.c4 button:focus,
.c4 button:hover {
  color: #FFFFFF;
}

.c4 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 a:focus,
.c4 a:hover {
  color: #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    float: left;
    margin-right: 3.5%;
    width: 22%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 18%;
    float: right;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 button {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 button {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

<div
  class="c0"
  dir="ltr"
  id="banner-id"
>
  <div
    class="c1"
    dir="ltr"
  >
    <h2
      class="c2"
      dir="ltr"
    >
      We've updated our Privacy and Cookies Policy
    </h2>
    <p>
      Hello
    </p>
    <ul
      class="c3"
      dir="ltr"
    >
      <li
        class="c4"
        dir="ltr"
      >
        <button
          type="button"
        >
          Accept
        </button>
      </li>
      <li
        class="c4"
        dir="ltr"
      >
        <a
          href="https://foobar.com"
        >
          Reject
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ConsentBanner should correctly render for rtl service 1`] = `
.c0 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  background-color: #323232;
  padding: 1rem 0.5rem;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c1::after {
  content: '\\0020';
  display: block;
  height: 0;
  clear: both;
  overflow: hidden;
  visibility: hidden;
}

.c2 {
  font-size: 1.375rem;
  line-height: 1.75rem;
  color: #FFFFFF;
  font-weight: 700;
  padding: 0;
  margin: 0;
}

.c3 {
  font-size: 1.0625rem;
  line-height: 1.5rem;
  color: #F6A21D;
  font-weight: 600;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c3 li + li {
  padding-top: 0.5rem;
}

.c4 button {
  font-size: 1.375rem;
  line-height: 1.75rem;
  color: #F6A21D;
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
}

.c4 button:focus,
.c4 button:hover {
  color: #FFFFFF;
}

.c4 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 a:focus,
.c4 a:hover {
  color: #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.625rem;
    line-height: 2.125rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    float: right;
    margin-left: 3.5%;
    width: 22%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 1.125rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 18%;
    float: left;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 button {
    font-size: 1.5rem;
    line-height: 2rem;
  }
}

@media (min-width:37.5rem) {
  .c4 button {
    font-size: 1.625rem;
    line-height: 2.125rem;
  }
}

<div
  class="c0"
  dir="rtl"
  id="banner-id"
>
  <div
    class="c1"
    dir="rtl"
  >
    <h2
      class="c2"
      dir="rtl"
    >
      عنوان
    </h2>
    <p>
      مرحبا
    </p>
    <ul
      class="c3"
      dir="rtl"
    >
      <li
        class="c4"
        dir="rtl"
      >
        <button
          type="button"
        >
          قبول
        </button>
      </li>
      <li
        class="c4"
        dir="rtl"
      >
        <a
          href="https://foobar.com"
        >
          رفض
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ConsentBanner with hidden attribute on wrapper should correctly render 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  background-color: #323232;
  padding: 1rem 0.5rem;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c1::after {
  content: '\\0020';
  display: block;
  height: 0;
  clear: both;
  overflow: hidden;
  visibility: hidden;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #FFFFFF;
  font-weight: 700;
  padding: 0;
  margin: 0;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #F6A21D;
  font-weight: 600;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c3 li + li {
  padding-top: 0.5rem;
}

.c4 button {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #F6A21D;
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
}

.c4 button:focus,
.c4 button:hover {
  color: #FFFFFF;
}

.c4 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 a:focus,
.c4 a:hover {
  color: #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    float: left;
    margin-right: 3.5%;
    width: 22%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    width: 18%;
    float: right;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 button {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c4 button {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

<div
  class="c0"
  dir="ltr"
  hidden=""
  id="banner-id"
>
  <div
    class="c1"
    dir="ltr"
  >
    <h2
      class="c2"
      dir="ltr"
    >
      We've updated our Privacy and Cookies Policy
    </h2>
    <p>
      Hello
    </p>
    <ul
      class="c3"
      dir="ltr"
    >
      <li
        class="c4"
        dir="ltr"
      >
        <button
          type="button"
        >
          Accept
        </button>
      </li>
      <li
        class="c4"
        dir="ltr"
      >
        <a
          href="https://foobar.com"
        >
          Reject
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ConsentBannerText should correctly render 1`] = `
.c0 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #BEBEBE;
}

.c0 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus,
.c0 a:hover {
  color: #FFFFFF;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    margin: 0;
    float: left;
    width: 53%;
  }
}

<p
  class="c0"
  dir="ltr"
>
  We have made some important changes to our Privacy and Cookies Policy and we want you to know what this means for you and your data.
</p>
`;
